lib/smooth/application/io/spi/BME280SPI.cpp

```diff
@@ -0,0 +1,256 @@
+        auto res = read(BME280Core::ID_REG, rxdata.data(), 2);
+
+        return res ? rxdata.at(1) : 0;
+    }
+
+    bool BME280SPI::reset()
+    {
+        std::vector<uint8_t> datagram;
+
+        // for spi write bit 7 of register address must be zero
+        // modify 0xE0 to 0x60;
+        datagram.push_back( BME280Core::RESET_REG& 0x7F );
+
+        // magic number to the reset device
+        datagram.push_back( 0xB6 );
+
+        return write(datagram);
+    }
+
+    bool BME280SPI::read_status(bool& is_measuring, bool& updating_from_nvm)
+    {
+        // The spi rx buffer need to be length of multiples of 32 bits to avoid heap corruption.
+        alignas(4) std::array<uint8_t, 4> rxdata;
+
+        auto res = read(BME280Core::STATUS_REG, rxdata.data(), 2);
+
+        if (res)
+        {
+            is_measuring = rxdata.at(1) & 0x4;
+            updating_from_nvm = rxdata.at(1) & 0x1;
+        }
+
+        return res;
+    }
+
+    bool BME280SPI::configure_sensor(BME280Core::SensorMode mode,
+                                     BME280Core::OverSampling humidity,
+                                     BME280Core::OverSampling pressure,
+                                     BME280Core::OverSampling temperature,
+                                     BME280Core::StandbyTimeMS standby,
+                                     BME280Core::FilterCoeff filter_coeff,
+                                     BME280Core::SpiInterface spi_interface)
+    {
+        std::vector<uint8_t> datagram;
+        bme280_core.get_configure_sensor_datagram(datagram,
+                                             mode,
+                                             humidity,
+                                             pressure,
+                                             temperature,
+                                             standby,
+                                             filter_coeff,
+                                             spi_interface);
+
+        // for spi write the bit 7 of register address must be zero
+        datagram.at(0) = datagram.at(0) & 0x7F;  // modify 0xF2 to 0x72
+        datagram.at(2) = datagram.at(2) & 0x7F;  // modify 0xF4 to 0x74
+        datagram.at(4) = datagram.at(4) & 0x7F;  // modify 0xF5 to 0x75
+
+        return write(datagram);
+    }
+
+    bool BME280SPI::read_configuration(BME280Core::SensorMode& mode,
+                                       BME280Core::OverSampling& humidity,
+                                       BME280Core::OverSampling& pressure,
+                                       BME280Core::OverSampling& temperature,
+                                       BME280Core::StandbyTimeMS& standby,
+                                       BME280Core::FilterCoeff& filter,
+                                       BME280Core::SpiInterface& spi_interface)
+    {
+        // The spi rx buffer need to be length of multiples of 32 bits to avoid heap corruption.c
```

Comment:
Trailing "c".

Is there a reference to point to re. the risk of heap corruption?

Reply:
Updated